regression-test/suites/function_p0/test_agg_foreach.groovy

```diff
@@ -0,0 +1,192 @@
+            "replication_allocation" = "tag.location.default: 1",
+            "storage_format" = "V2"
+    );
+    """
+    sql """
+   insert into table_with_null values(1,[1,2,3],["ab","123"]),(2,[20],["cd"]),(3,[100],["efg"]) , (4,null,null),(5,[null,2],[null,'c']);
+   """
+
+
+    qt_sql """
+       select sum_foreach(a) ,  count_foreach(a) , group_concat_foreach(s)  from table_with_null;
+    """
+
+
+    // for nereids_planner
+
+    sql """ set enable_nereids_planner=true;"""
+    sql """  set enable_fallback_to_original_planner=false;"""
+   
+    sql """
+        drop table if exists table_with_all_null;
+    """
+    sql """
+    CREATE TABLE IF NOT EXISTS table_with_all_null (
+              `id` INT(11) NULL COMMENT "",
+              `a` array<INT> NULL COMMENT "",
+              `s` array<String>  NULL COMMENT ""
+            ) ENGINE=OLAP
+            DUPLICATE KEY(`id`)
+            DISTRIBUTED BY HASH(`id`) BUCKETS 1
+            PROPERTIES (
+            "replication_allocation" = "tag.location.default: 1",
+            "storage_format" = "V2"
+    );
+    """
+    sql """
+       insert into table_with_all_null values
+        (1,null,null),
+        (2,null,null),
+        (3,null,null);
+    """
+
+    qt_sql """ 
+      select sum_foreach(a) ,  count_foreach(a) , group_concat_foreach(s)   from table_with_all_null;
+    """
+
+    sql """
+        drop table if exists table_with_all_not_null;
+    """
+
+     sql """
+        CREATE TABLE IF NOT EXISTS table_with_all_not_null (
+              `id` INT(11)  COMMENT "",
+              `a` array<INT>  COMMENT "",
+              `s` array<String>   COMMENT ""
+            ) ENGINE=OLAP
+            DUPLICATE KEY(`id`)
+            DISTRIBUTED BY HASH(`id`) BUCKETS 1
+            PROPERTIES (
+            "replication_allocation" = "tag.location.default: 1",
+            "storage_format" = "V2"
+    );
+    """
+    
+    sql """
+    insert into table_with_all_not_null values(1,[1,2,3],["ab","123"]),(2,[20],["cd"]),(3,[100],["efg"]);
+    """
+
+    qt_sql """
+       select sum_foreach(a) ,  count_foreach(a) , group_concat_foreach(s)  
+        from table_with_all_not_null;
+    """
+
+    sql """
+       drop table if exists table_with_null;
+    """
+
+    sql """
+       CREATE TABLE IF NOT EXISTS table_with_null (
+              `id` INT(11)  COMMENT "",
+              `a` array<INT>  COMMENT "",
+              `s` array<String>   COMMENT ""
+            ) ENGINE=OLAP
+            DUPLICATE KEY(`id`)
+            DISTRIBUTED BY HASH(`id`) BUCKETS 1
+            PROPERTIES (
+            "replication_allocation" = "tag.location.default: 1",
+            "storage_format" = "V2"
+    );
+    """
+    sql """
+    insert into table_with_null values(1,[1,2,3],["ab","123"]),(2,[20],["cd"]),(3,[100],["efg"]) , (4,null,null),(5,[null,2],[null,'c']);
+   """
+
+
+    qt_sql """
+       select sum_foreach(a) ,  count_foreach(a) , group_concat_foreach(s) from table_with_null;
```

Comment:
if all agg function could have foreach suffix. u should add all agg function with foreach suffix tests